### ISSUE TITLE
Anova1rm - correction for post-hoc testing 

### ISSUE THREAD

**Issue by ThomasBraeckevelt** [0todd0000/spm1dmatlab#127]
Hello !

We performed a biomechanical analysis on 14 shoulders, doing 6 trials of a certain movement in 5 progressively worse (ligaments cut) conditions. Then we selected the 3 best trials, time-normalized the data and calculated the mean value for the 101 data points of these trials. As such we have one observation per movement per condition per shoulder. By means of a one-way repeated measures anova, we compared the different conditions.
We use 7 post-hoc t-tests. 

- Is the bonferroni correction appropriate to use in our case?
- How should we appropriately correct for this post-hoc analysis? Should we manually correct the alpha (picture below), in this case 0.05/7=0.00714... Or is this already accounted for in the script?
![Knipsel](https://user-images.githubusercontent.com/63154136/86541040-e96bfa00-bf09-11ea-9405-4bbff45842c8.PNG)
![image](https://user-images.githubusercontent.com/63154136/86541108-86c72e00-bf0a-11ea-8e9d-199630923a0c.png)

- Furthermore i have understood that every p-value that we received after the anova1rm (in this example: 0.0167 0.0469 0.0452 0.0073 0.0065) shoulde be corrected using the following scriptlines. Is this correct?
![image](https://user-images.githubusercontent.com/63154136/86541323-3b158400-bf0c-11ea-83bb-a1a0bcfbc3b7.png)
![image](https://user-images.githubusercontent.com/63154136/86541270-c6dae080-bf0b-11ea-8b36-600ae4fb59d9.png)
![image](https://user-images.githubusercontent.com/63154136/86541258-b460a700-bf0b-11ea-8a33-29af631716e3.png)

Thank you very much for your help!

**Comment by 0todd0000** [comment]
Hello!

> Is the bonferroni correction appropriate to use in our case?

Appropriateness is difficult to judge, but relevant to appropriateness is the following:
The Bonferroni correction is the most conservative correction. It assumes independence, but this assumption is almost always overly conservative in practical experimentation. For *post hoc* analysis, the Bonferroni correction is fine. However, caution must be exercised if *post hoc* results disagree with the main test, where the main test is the one that best describes the whole experiment, in this case the RM-ANOVA. There are procedures for achieving more exact agreement between *post hoc* tests and the main test, but these procedures are not implemented in **spm1d**.

> How should we appropriately correct for this post-hoc analysis? Should we manually correct the alpha (picture below), in this case 0.05/7=0.00714... Or is this already accounted for in the script?

This is correct.  Single calls to **spm1d.stats** functions like **ttest_paried** presume that this is the only test in the experiment. Modeling an experiment using multiple tests requires a correction across those tests, and the correction you have used here looks fine.


> Furthermore i have understood that every p-value that we received after the anova1rm (in this example: 0.0167 0.0469 0.0452 0.0073 0.0065) shoulde be corrected using the following scriptlines. Is this correct?

Yes, that its correct. The code looks good.

**Comment by ThomasBraeckevelt** [comment]
Thank you for the swift reply Todd!

Just to be sure i have understood correctly, the code should look like this when correcting for 7 post-hoc t-tests?
_alpha      = 0.00714285714285714285714285714286;
nTests     = 7;
p_critical = spm1d.util.p_critical_bonf(alpha, nTests)
t01i       = t01.inference(p_critical, 'two_tailed',true)
t02i       = t02.inference(p_critical, 'two_tailed',true)
t03i       = t03.inference(p_critical, 'two_tailed',true)
t04i       = t04.inference(p_critical, 'two_tailed',true)
t23i       = t23.inference(p_critical, 'two_tailed',true)
t24i       = t24.inference(p_critical, 'two_tailed',true)
t34i       = t34.inference(p_critical, 'two_tailed',true)_

I have one more question regarding the visualisation of SPM{t} curves.
In the following study, because of the high number of statistical analyses, the SPM results are presented in a summarized manner. Instead of SPM{t} curves, black bars are shown, indicating significance during the gait cycle.
I was wondering if you could help us with some guidance how to visualise our SPM {t} curves in a similar/identical fashion?
_Citation: Nieuwenhuys A, Papageorgiou E, Desloovere K, Molenaers G, De Laet T (2017) Statistical Parametric Mapping to Identify Differences between Consensus-Based Joint Patterns during Gait in Children with Cerebral Palsy. PLOS ONE 12(1): e0169834_
![image](https://user-images.githubusercontent.com/63154136/86601527-bbd28f80-bfa1-11ea-9a0c-53d22a183247.png)
![image](https://user-images.githubusercontent.com/63154136/86600887-f38d0780-bfa0-11ea-8b02-42d5922a22c9.png)

Thank you very much!

**Comment by 0todd0000** [comment]
Hi, for the corrected thresholds use alpha=0.05 like this:

```matlab
alpha = 0.05;
nTests = 7;
p_critical = spm1d.util.p_critical_bonf(alpha, nTests)
```



>I have one more question regarding the visualisation of SPM{t} curves.
>In the following study, because of the high number of statistical analyses, the SPM results are presented in a summarized manner. Instead of SPM{t} curves, black bars are shown, indicating significance during the gait cycle.
I was wondering if you could help us with some guidance how to visualise our SPM {t} curves in a similar/identical fashion?
Citation: Nieuwenhuys A, Papageorgiou E, Desloovere K, Molenaers G, De Laet T (2017) Statistical Parametric Mapping to Identify Differences between Consensus-Based Joint Patterns during Gait in Children with Cerebral Palsy. PLOS ONE 12(1): e0169834

This is indeed a nice visualization, but this requires custom coding that is not available in **spm1d**, so we are unfortunately unable to provide support for this.
